### PR TITLE
[perf-counters] Fix pause/resume

### PR DESCRIPTION
**Pull request by mtrofin**
Using `state.PauseTiming() / state.ResumeTiming()` was broken.

Thanks @virajbshah for the the repro testcase.

### CODE REVIEW

File: src/benchmark.cc [hunk 1]
```diff
@@ -230,7 +230,7 @@ void State::PauseTiming() {
       auto name = name_and_measurement.first;
       auto measurement = name_and_measurement.second;
       BM_CHECK_EQ(std::fpclassify(double{counters[name]}), FP_ZERO);
```
Comment (LebedevRI) [inline src/benchmark.cc:232]: I don't believe this assertion remains valid after this change.

Comment (mtrofin) [reply src/benchmark.cc]: thanks for the catch - in a debug build it trips, indeed.

File: test/perf_counters_test.cc [hunk 2]
```diff
@@ -21,17 +21,62 @@ static void BM_Simple(benchmark::State& state) {
...
+
+ADD_CASES(TC_JSONOut, {{"\"name\": \"BM_WithPauseResume\",$"}});
+
 static void CheckSimple(Results const& e) {
   CHECK_COUNTER_VALUE(e, double, "CYCLES", GT, 0);
   CHECK_COUNTER_VALUE(e, double, "BRANCHES", GT, 0.0);
 }
+
+static void CheckInstrCount(Results const& e) {
+  CHECK_COUNTER_VALUE(e, double, "INSTRUCTIONS", GT, kIters);
```
Comment (dmah42) [inline test/perf_counters_test.cc:68]: with pause and resume, shouldn't this be GT 2*kIters?  i don't know how the INSTRUCTIONS perf counters work so i could be wrong.

Comment (mtrofin) [reply test/perf_counters_test.cc:68]: Yes - the current test checks against the original regression, but I see what you mean. Ideally we'd check that the value in the "with pause resume" is < 2 * "value in the without pause resume". Is there a recommended way to do that, other than capturing the value in the "without pause resume" in some global and relying on execution order?

Comment (dmah42) [reply test/perf_counters_test.cc:68]: would `LT, 1.5*kIters` not be enough at least for an upper bound?

Comment (mtrofin) [reply test/perf_counters_test.cc:68]: each iteration executes a bit of instructions, and it depends on opt level how many (and architecture)

Comment (dmah42) [reply test/perf_counters_test.cc:68]: we can't rely on execution order.  we can guarantee that both have run when `RunOutputTests` completes, so maybe we can store the number of instructions for both BMs and compare them?

Comment (mtrofin) [reply test/perf_counters_test.cc]: done.

**Review by LebedevRI — APPROVED** [review]
The test is rather fragile, but seems fine in general.
Thank you.

**Comment by dmah42** [comment]
thanks!

**Pull request merged**